### PR TITLE
COPS-626 Exponential backoff

### PR DESCRIPTION
**Pull request by marty331**
- Utilize exponential backoff for all calls to Stripe
- Implement Tenacity to handle exponential backoff
- Implement use of indempotency key for Stripe Post calls
- Positive and negative testing of all

### CODE REVIEW

File: src/shared/universal.py [hunk 1]
```diff
@@ -17,3 +19,7 @@ def format_plan_nickname(product_name: str, plan_interval: str) -> str:
         formatted_interval = plan_interval
 
     return f"{product_name} ({formatted_interval})".title()
+
+
+def get_indempotency_key() -> str:
```
Comment (bowlofstew) [inline src/shared/universal.py]: We will need to re-run performance around some of these cases.  This calls has performance implications.

Comment (marty331) [reply src/shared/universal.py]: So are you going to re-run perfomance? Or are you going to teach me how?  I'm good either way.

Comment (bowlofstew) [reply src/shared/universal.py]: After this lands, we can walk thru how to do it.

File: src/shared/vendor.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+@retry(
+    wait=wait_exponential(multiplier=1, min=1, max=8),
+    stop=stop_after_attempt(4),
+    reraise=True,
+)
+def build_stripe_subscription(
+    customer_id: str, plan_id: str, idempotency_key: str
+) -> Subscription:
+    """
+    Create a new Stripe subscription for a given customer
```
Comment (bowlofstew) [inline src/shared/vendor.py]: The only thing to note in documentation is that it appears to be inconsistent between these methods and others below.

**Pull request merged**